File: problem52/Problem52.java
import java.util.*; 
/* Problem 52 */

public class Problem52 {
	public static void main(String[] arg) {
		int i = 1; int n = 6;
		int k;
		while (true) {
			for (k = 2; k < n; k++) {
				if (!Combinatorics.isPerm(i, i * k)) break;
			}
			if (Combinatorics.isPerm(i, i * k) && k == n) {
				System.out.println("The solution is " + i);
				break;
			}
			i++;
		}
	}
}

File: problem52/problem52.js
/* Problem 52 */

var Combinatorics = require("../../Documents/projet_personnel/informatique/javaScript/Combinatorics.js");
var i = 1; var n = 6;
var k = 0;
while (true) {
	for (k = 2; k < n; k++) {
		if (!Combinatorics.isPerm(i, i * k)) break;
	}
	if (Combinatorics.isPerm(i, i * k) && k === n) {
		console.log("The solution is " + i);
		break;
	}
	i++;
}
